discourse-doctor

```diff
@@ -0,0 +1,374 @@
+	echo This suggests that your DNS settings are correct, | tee -a $LOG_FILE
+	echo but something is keeping traffic to port 443 from getting to your server. | tee -a $LOG_FILE
+	echo Check your networking configuration to see that connections to port 443 are allowed. | tee -a $LOG_FILE
+    else
+      echo "A connection to http://$HOST (port 80) also fails." | tee -a $LOG_FILE
+      echo | tee -a $LOG_FILE
+      echo This suggests that $HOST resolves to the wrong IP address | tee -a $LOG_FILE
+      echo or that traffic is not being routed to your server. | tee -a $LOG_FILE
+    fi
+    echo | tee -a $LOG_FILE
+    echo Google: \"open ports YOUR CLOUD SERVICE\" for information for resolving this problem. | tee -a $LOG_FILE
+    echo | tee -a $LOG_FILE
+    echo This test might not work for all situations, | tee -a $LOG_FILE
+    echo so if you can access Discourse at http://$HOST, this might not indicate a problem. | tee -a $LOG_FILE
+    sleep 3
+  fi
+}
+
+##
+## Do we have docker?
+##
+check_docker_version () {
+  echo  -e "\n\n==================== DOCKER INFO ====================" | tee -a $LOG_FILE
+  docker_path=`which docker.io || which docker`
+  if [ -z $docker_path ]; then
+    echo "Docker is not installed. Have you installed Discourse at all?"
+    echo "Perhaps you're looking for ./discourse-setup ."
+    echo "There is no point in continuing."
+    exit
+  else
+    echo -n "DOCKER VERSION: " | tee -a $LOG_FILE
+    docker --version | tee -a  $LOG_FILE
+    echo -e "\nDOCKER PROCESSES" | tee -a $LOG_FILE
+    sudo docker ps -a | tee -a $LOG_FILE
+  fi
+}
+
+check_OS() {
+  echo "----------------------------------------"
+  echo -n "OS:"  |tee -a $LOG_FILE
+  echo `uname -s` |tee -a  $LOG_FILE
+}
+
+check_osx_memory() {
+  echo "----------------------------------------"
+  echo -n "OSX RAM:" | tee -a $LOG_FILE
+  free -m | awk '/Mem:/ {print $2}' | tee -a  $LOG_FILE
+  free -m | tee -a $LOG_FILE
+}
+
+check_linux_memory() {
+  echo "----------------------------------------"
+  echo -n "RAM (MB):" |tee -a $LOG_FILE
+  free -m --si | awk ' /Mem:/  {print $2} ' | tee -a $LOG_FILE
+  free -m | tee -a $LOG_FILE
+}
+
+check_disk_and_memory() {
+
+  os_type=$(check_OS)
+  if [ "$os_type" == "Darwin" ]; then
+    check_osx_memory
+  else
+    check_linux_memory
+  fi
+
+  echo -e "\n\n==================== DISK SPACE CHECK ====================" | tee -a $LOG_FILE
+  echo System Disk Space | tee -a $LOG_FILE
+  df -h / /var/discourse /var/lib/docker /var/lib/docker/* |uniq | tee -a  $LOG_FILE
+
+  if [ "$version" != "NOT FOUND" ]
+  then
+    echo | tee -a $LOG_FILE
+    echo Container Disk Space | tee -a $LOG_FILE
+    sudo docker exec -w /var/www/discourse -i $app_name df -h / /shared/ /shared/postgres_data /shared/redis_data /shared/backups /var/log|uniq | tee -a  $LOG_FILE
+  fi
+
+
+  echo -e "\n\n==================== DISK INFORMATION ====================" >> $LOG_FILE
+  sudo fdisk -l >> $LOG_FILE
```

Comment:
This might be problematic https://github.com/koalaman/shellcheck/wiki/SC2024

Reply:
No. It can't be problematic, because they have to run the script as root, but it is superfluous.